### ISSUE TITLE
Add option to enable Firebug

### ISSUE THREAD

**Issue by codewithmichael** [codewithmichael/babel-browser-harness#3]
Development via mobile device is one of the  driving factors for BBH, and mobile browsers don't typically have the same debugging capabilities that desktop browsers do, often not even having a viewable console.

As Firebug is already mentioned as a useful workaround in the FAQ, we might as well have an option to enable it from within BBH.
#### Implementation

It's kind of the same concept as the default modules, but in this case the script tag below needs to be added to the document immediately instead of waiting for `execute()`, otherwise some of the messages will be missed.

``` html
<script src="https://getfirebug.com/firebug-lite.js#startOpened"></script>
```

**Comment by codewithmichael** [comment]
Also, when using Firebug Lite, some JS errors are not processed through the console. It still received the `BBH ♥ Error Detected :(` message, but the actual error is not logged.

One way to fix this would be to explicitly log the error in `logAndThrowError()`. Of course, in a real browser console this would mean the error gets logged twice, so it should probably be an option.

**Comment by codewithmichael** [comment]
PR #20, which handles autoloading React for issue #10, already has a useful autoloading method as well as a constant to store autoloadable libraries, so I'm going to base the code for this on the `issue10` branch, which is currently in review and will likely make it into an upcoming release.

If for some reason issue #10 (or PR #20) doesn't make it, I'll extract the useful parts out later.

**Comment by codewithmichael** [comment]
This has been trickier than expected, but I got it working—commit: 6fbf2c8.

As expected, it depends on the `issue10` branch.

I added a variable called `executionDelays` that holds a list of promises used to delay calling `execute()`.  The new method, `enableFirebug()`, adds a promise to `executionDelays` that waits for the created script tag to load or error. It then resolves or rejects accordingly.

For now, `executionDelays` catches all errors and just logs them, so if the Firebug Lite script fails to load, BBH will still continue as usual.

I still need to update the documentation before this will be ready for review.

**Comment by codewithmichael** [comment]
It seems that Firebug Lite defines/redefines `window.XMLHttpRequest` when the current browser is not IE6.

In mobile testing—Working Copy (iOS)—BBH reports `XMLHttpRequest` as a global leak. Perhaps `file://`-based Webkit instances don't always declare `XMLHttpRequest`?

In any case, I've added `"XMLHttpRequest"` to the `ignores` list for `firebug-lite`. If it's already defined it makes no difference at all.

commit: b66910e

**Comment by codewithmichael** [comment]
This issue is resolved by PR #21, which is now in review. Marking this issue also in review.

Note that PR #21 depends on PR #20, and therefore issue #10.

**Comment by codewithmichael** [comment]
**Note:** Problems with Firebug Lite, as mentioned in [the first comment](#issuecomment-229698072),  aren't covered by PR #20, and won't be covered by this ticket.

While it's worthwhile to make the debugging process as smooth as possible, it's shaky territory to start debugging the debug console this soon after making it available—it's still possible we'll select another console option.

Any issues with Firebug Lite not catching certain types of errors should be posted in a separate issue.
